### DISCUSSION TITLE
error while generating HYMAP2 routing parameters with LDT

### DISCUSSION THREAD

**Discussion by zhengzhenk** [NASA-LIS/LISF#1444] [Q&A] [answered]
I am currently attempting to run the HYMAP2 routing module in LISF v7.4 and Noah-MP v4.0.1. I downloaded HYMAP2 routing parameter files at 0.125° resolution from the LIS data portal website: 
`https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/HYMAP_parms/NLDAS_12.5KM/`.

However, after using LDT to generate LIS input files, I noticed that all variables related to HYMAP are set to 0. Might the issue with zero values be related to the boundaries and resolution settings in the ldt.config for HYMAP parameters (as shown below)?
`HYMAP params spatial transform:   none #average #budget-bilinear # average # none`
`HYMAP params map projection:   latlon`
`HYMAP params lower left lat:   -59.9375`
`HYMAP params lower left lon:   -179.9375`
`HYMAP params upper right lat:   89.9375`
`HYMAP params upper right lon:   179.9375`
`HYMAP params resolution (dx):   0.125 # 0.125`
`HYMAP params resolution (dy):   0.125 # 0.125`

In the meantime, when I used HYMAP2 routing parameter files at 0.25° resolution from the LIS data portal website: `https://portal.nccs.nasa.gov/lisdata_pub/data/PARAMETERS/HYMAP_parms/HYMAP_25KM/` ,
LDT was able to generate the correct LIS input files using these parameter files, and LIS ran the HYMAP2 routing module correctly.

I have attached my ldt.config, ldtlog, and lis.input files below. Any suggestions would be appreciated.
@dmocko , @agetiran 

Thank you!

[ldt_hymap_0125_test.zip](https://github.com/NASA-LIS/LISF/files/13266741/ldt_hymap_0125_test.zip)

**Comment by dmocko** [comment, marked as answer]
Hi @zhengzhenk 

As your log file clearly states: `the parameter and LIS grid resolutions must be the same.`

You have set the dx/dy for the HYMAP params at 0.125, while your have set your Run domain at 0.05.

The NLDAS 12.5-km parameters will only work for the NLDAS domain, which is defined here:
https://ldas.gsfc.nasa.gov/nldas/specifications

Thus, for these parameters, these config entries must be defined as:

```
HYMAP params lower left lat:              25.0625
HYMAP params lower left lon:              -124.9375
HYMAP params upper right lat:             52.9375
HYMAP params upper right lon:             -67.0625
```

But the main issue is that this parameters will only work within this domain.
Your Run domain is trying to use data outside of the NLDAS region, which
is primarily for CONUS.

For your domain, you will need to use the 0.25-degree global parameters.

If you have a collaboration with us, you may be able to request parameters
at a finer resolution for your domain.

**Reply by zhengzhenk** [reply]
Hi @dmocko 
Thank you for the explanation of the NLDAS 12.5-KM parameter. I have not cooperated with you for the time being, but I sincerely hope to have the opportunity to cooperate with you in the future. I will be using the global parameters at 0.25° resolution.
Thanks again！
